### PR TITLE
apng2gif, gif2apng, apngopt: init

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Zaedus** [comment]
Result of `nixpkgs-review pr 223931` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>apng2gif</li>
    <li>apngopt</li>
    <li>gif2apng</li>
  </ul>
</details>

File: pkgs/tools/graphics/apngopt/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  buildInputs = [
+    libpng
+    zlib
+    zopfli
+  ];
+
+  preBuild = ''
+    buildFlagsArray+=("LIBS=-lzopfli -lstdc++ -lpng -lz")
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/apngopt/default.nix:36]: buildFlags does not work here?

File: pkgs/tools/graphics/gif2apng/default.nix [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  src = fetchzip {
+    url = "mirror://sourceforge/gif2apng/gif2apng-${version}-src.zip";
+    stripRoot = false;
+    hash = "sha256-rt1Vp4hjeFAVWJOU04BdU2YvBwECe9Q1c7EpNpIN+uE=";
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/g/gif2apng/1.9%2Bsrconly-3%2Bdeb11u1/debian/patches/10-7z.patch";
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/gif2apng/default.nix:21]: I think those links are not save to use because they disappear when new versions are released

Comment (fgaz) [reply pkgs/tools/graphics/gif2apng/default.nix:21]: It should be fine, I see all versions at https://sources.debian.org/data/main/g/gif2apng/

File: pkgs/tools/graphics/apngopt/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/tools/graphics/apngopt/default.nix:36]: No, there are spaces in the argument

**Comment by fgaz** [comment]
Fixed darwin

File: pkgs/tools/graphics/gif2apng/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/graphics/gif2apng/default.nix:21]: Those are only the versions that are in some current release series. See https://github.com/NixOS/nixpkgs/commit/fb7c2cdb519dcffb3cd3b29b8179a306afaecd74 and https://github.com/NixOS/nixpkgs/blob/2341c80301d63eecb06f41b69b3cd1bee2f243b9/pkgs/development/libraries/libconfuse/default.nix#L25

**Comment by pinpox** [comment]
@fgaz any news on this?

**Comment by SuperSandro2000** [comment]
@ofborg build apng2gif gif2apng apngopt

**Pull request merged**